arithmetization/src/test/java/net/consensys/linea/zktracer/exceptions/OutOfGasMemExpExceptionTest.java

```diff
@@ -0,0 +1,604 @@
+            .build();
+
+    program
+        // 1. Create contract - returns 32FF
+        .push(codeOwnerAccount.getAddress())
+        .op(OpCode.EXTCODESIZE)
+        .op(OpCode.DUP1)
+        .push(0)
+        .push(0)
+        .push(codeOwnerAccount.getAddress())
+        .op(OpCode.EXTCODECOPY)
+        .push(0)
+        .push(0)
+        .op(OpCode.CREATE)
+        // 2. Execute static call
+        .push(0) // byte size of return data
+        .push(0) // retOffset
+        .push(0) // byte size calldata
+        .push(0) // argsOffset
+        .op(OpCode.DUP5) // Address of the contract deployed above
+        .push(Bytes.fromHexString("0xFFFFFFFF")) // gas
+        .op(OpCode.STATICCALL)
+        // 3. Clean the stack
+        .op(OpCode.POP)
+        .op(OpCode.POP)
+        // 4. Return data copy
+        .push(32) // size
+        .push(0) // offset
+        .push(65) // destoffset, trigger mem expansion
+        .op(OpCode.RETURNDATACOPY);
+
+    Bytes pgCompile = program.compile();
+    BytecodeRunner bytecodeRunner = BytecodeRunner.of(pgCompile);
+
+    long gasCost = bytecodeRunner.runOnlyForGasCost(List.of(codeOwnerAccount));
+
+    bytecodeRunner.run(gasCost + cornerCase, List.of(codeOwnerAccount));
+    if (cornerCase == -1) {
+      assertEquals(
+          OUT_OF_GAS_EXCEPTION,
+          bytecodeRunner.getHub().previousTraceSection().commonValues.tracedException());
+    } else {
+      assertNotEquals(
+          OUT_OF_GAS_EXCEPTION,
+          bytecodeRunner.getHub().previousTraceSection().commonValues.tracedException());
+    }
+  }
+
+  @ParameterizedTest
+  @ValueSource(ints = {-1, 0, 1})
+  void outOfGasExceptionRevert(int cornerCase) {
+    BytecodeCompiler program = BytecodeCompiler.newProgram();
+
+    program
+        .push(
+            Bytes.fromHexString(
+                "0xF00000000000000000000000000000000000000000000000000000000000FFAA")) // value
+        .push(0) // offset
+        .op(OpCode.MSTORE)
+        .push(3) // value
+        .push(30) // offset
+        .op(OpCode.REVERT);
+
+    Bytes pgCompile = program.compile();
+    BytecodeRunner bytecodeRunner = BytecodeRunner.of(pgCompile);
+
+    long gasCost = bytecodeRunner.runOnlyForGasCost();
+
+    bytecodeRunner.run(gasCost + cornerCase);
+    if (cornerCase == -1) {
+      assertEquals(
+          OUT_OF_GAS_EXCEPTION,
+          bytecodeRunner.getHub().previousTraceSection().commonValues.tracedException());
+    } else {
+      assertNotEquals(
+          OUT_OF_GAS_EXCEPTION,
+          bytecodeRunner.getHub().previousTraceSection().commonValues.tracedException());
+    }
+  }
+
+  @ParameterizedTest
+  @ValueSource(ints = {-6420, -6419, -6418, 100, 101, 102})
+  /*
+  Deployment code: "0x7F7EFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF60005260206000F3"
```

Comment:
What's the reason for using a `PUSH31` rather than a `PUSH32` as the first operation of the deployed code ? Just asking to understand what is going on. It seems to me that the code you are deploying is
```go
0x007effff..ff
```
I.e.
```rust
STOP
PUSH31 0xff .. ff
```
So that the deployed code, when called, will `STOP` immediately.

Reply:
I guess it's not being called so it doesn't matter. Still an odd choice IIUC.

Reply:
My bad, I got mixed up. The deployed code is
```rust
PUSH31 0xff .. ff
```
no `STOP`.